pkg/collector/corechecks/gpu/gpu.go

```
@@ -0,0 +1,206 @@
+// Cancel cancels the check
+func (m *Check) Cancel() {
+	ret := nvml.Shutdown()
+	if ret != nvml.SUCCESS {
+		log.Warnf("Failed to shutdown NVML: %v", nvml.ErrorString(ret))
+	}
+}
+
+// Configure parses the check configuration and init the check
+func (m *Check) Configure(senderManager sender.SenderManager, _ uint64, config, initConfig integration.Data, source string) error {
+	if err := m.CommonConfigure(senderManager, initConfig, config, source); err != nil {
+		return err
+	}
+	if err := m.config.Parse(config); err != nil {
+		return fmt.Errorf("ebpf check config: %s", err)
+	}
+
+	return nil
+}
+
+func (m *Check) ensureInitialized() error {
+	var err error
+
+	if m.sysProbeUtil == nil {
+		m.sysProbeUtil, err = processnet.GetRemoteSystemProbeUtil(
+			pkgconfigsetup.SystemProbe().GetString("system_probe_config.sysprobe_socket"),
+		)
+		if err != nil {
+			return fmt.Errorf("sysprobe connection: %s", err)
+		}
+	}
+
+	if m.timeResolver == nil {
+		m.timeResolver, err = sectime.NewResolver()
+		if err != nil {
+			return fmt.Errorf("cannot create time resolver: %s", err)
+		}
+	}
+	return nil
+}
+
+// ensureProcessor ensures that there is a stats processor for the given key
+func (m *Check) ensureProcessor(key *model.StreamKey, snd sender.Sender, gpuThreads int, checkDuration time.Duration) {
+	if _, ok := m.statProcessors[key.Pid]; !ok {
+		m.statProcessors[key.Pid] = &StatsProcessor{
+			key: key,
+		}
+	}
+
+	m.statProcessors[key.Pid].totalThreadSecondsUsed = 0
+	m.statProcessors[key.Pid].sender = snd
+	m.statProcessors[key.Pid].gpuMaxThreads = gpuThreads
+	m.statProcessors[key.Pid].measuredInterval = checkDuration
+	m.statProcessors[key.Pid].timeResolver = m.timeResolver
+	m.statProcessors[key.Pid].lastCheck = m.lastCheckTime
+}
+
+// Run executes the check
+func (m *Check) Run() error {
+	if err := m.ensureInitialized(); err != nil {
+		return err
+	}
+
+	gpuDevices, err := getGPUDevices()
+	if err != nil {
+		return fmt.Errorf("get GPU devices: %s", err)
```

i think you can return the error as is, without extra wrap